File: ui/task-tracker/src/app/shared/model/timeTrackingModel.ts
import { AutoTrackTimeStatus } from "../enums/auto-track-time-status";

export class TimeTrackingModel {
    /**
     * Затраченное время
     */
    timeSpent: string = '';

    /**
     * Дата начала работы
     */
    dateBegin!: string; // Не требует инициализации, так как не может быть null

    /**
     * Комментарий к списанным часам
     */
    comment?: string;

    /**
     * Статус автоматического трекинга времени.
     * Если не задано - значит трекинг выполнен руками
     */
    autoTrackStatus?: AutoTrackTimeStatus;

    /**
     * Юзер, который списал часы
     */
    userId!: number;

    issueId!: number;
}

File: TaskTracker.Core/src/Models/ResponseModels/TimeTrackingModel.cs
using TaskTracker.Core.src.Enums;

namespace TaskTracker.Core.src.Models.ResponseModels
{
    public class TimeTrackingModel
    {
        public int Id { get; set; }
        /// <summary>
        /// Затраченное время
        /// </summary>
        public string TimeSpent { get; set; } = string.Empty;

        /// <summary>
        /// Дата начала работы
        /// </summary>
        public string DateBegin { get; set; } = null!;

        /// <summary>
        /// Комментарий к списанным часам
        /// </summary>
        public string? Comment { get; set; }

        /// <summary>
        /// Статус автоматического трекинга времени. 
        /// Если не задано - значит трекинг выполнен руками
        /// </summary>
        public AutoTrackTimeStatus? AutoTrackStatus { get; set; }

        /// <summary>
        /// Юзер, который списал часы
        /// </summary>
        public int UserId { get; set; }

        public int IssueId { get; set; }
    }
}
